docs/cookbook/cli/handle-multiple-channels-in-cli.rst

```diff
@@ -0,0 +1,169 @@
+Handle multiple channels in CLI
+===============================
+
+When we use directly or indirectly any service depending on a channel context, we are not able to define which channel we want to use, when there are more than two.
+Your primary goal should be to avoid such cases, but if you have to, there is a way to do it.
+
+What is our goal?
+-----------------
+
+To accomplish this task, we have to create a custom channel context available only from the CLI context. But this concrete channel context cannot only return a channel,
+but also has to allow to pass a channel code as an argument (which we will get from the console command). This way, we can use the channel context in our services.
+
+1. Custom Channel Context
+-------------------------
+
+First, we need to create a custom channel context. Let's remind our requirements:
+
+    * available only from the CLI
+    * there must be away to pass in a channel code
+
+Our suggested solution is the following:
+
+.. code-block:: php
+
+    // src/Channel/Context/CliBasedChannelContext.php
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Channel\Context;
+
+
+    use Sylius\Component\Channel\Context\ChannelNotFoundException;
+    use Sylius\Component\Channel\Model\ChannelInterface;
+    use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+
+    final class CliBasedChannelContext implements CliBasedChannelContextInterface
+    {
+        private ?string $channelCode = null;
+
+        public function __construct(
+            private ChannelRepositoryInterface $channelRepository,
+        ) {
+        }
+
+        public function setChannelCode(?string $channelCode): void
+        {
+            $this->channelCode = $channelCode;
+        }
+
+        public function getChannelCode(): ?string
+        {
+            return $this-channelCode;
+        }
+
+        public function getChannel(): ChannelInterface
+        {
+            if ('cli' !== PHP_SAPI || null === $this->channelCode) {
```

Comment:
```suggestion
            if ('cli' !== \PHP_SAPI || null === $this->channelCode) {
```

Reply:
Why do you need \ before constant :D? The possibility of another PHP_SAPI in the same namespace what the class is minimal 🤔.

Reply:
The main reason is that this is a part of our coding standard :P
https://github.com/SyliusLabs/CodingStandard/blob/master/ecs.php#L164

And I also believe backslash should work faster as the interpreter knows exactly where the const is defined. So it should skip looking for it in our code.

Reply:
As this is only an example, and differences are minimal (and we agreed on Slack) I'm resolving it :D.

Reply:
![image](https://user-images.githubusercontent.com/13316080/209161060-96b2e8a2-684f-45cf-b84a-153b09308863.png)

Reply:
xD